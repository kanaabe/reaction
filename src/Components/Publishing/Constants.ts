import moment from "moment-timezone"

/**
 * Matches for Email / Instant Articles
 */
export const EMAIL_REGEX = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

/**
 * The quality to request from image CDN
 */
export const GLOBAL_IMAGE_QUALITY = 80

/**
 * TODO: Eventually remove sizeMe
 */
export const SIZE_ME_REFRESH_RATE = 500

/**
 * Relative path to article
 */
export const getArticleHref = slug => `/article/${slug}`

/**
 * Absolute path to article
 */
export const getArticleFullHref = slug => `https://www.artsy.net/article/${slug}`

/**
 * Relative path to editorial entity
 */
export const getEditorialHref = (type, slug) => `/${type}/${slug}`

/**
 * Absolute path to editorial entity
 */
export const getFullEditorialHref = (type, slug) => `https://www.artsy.net/${type}/${slug}`

/**
 * ByLine helpers
 * TODO: Move this into some kind of utils folder
 */
export const getAuthorByline = authors => {
  const authorCount = Number(authors && authors.length)

  if (authorCount === 1) {
    return authors[0].name || ''
  } else if (authorCount > 1) {
    const names = authors.reduce((prev, curr, i) => {
      let delim
      const len = authors.length
      if (i === len - 1) {
        delim = " and "
      } else if (i === 0) {
        delim = ""
      } else {
        delim = ", "
      }
      return prev + delim + curr.name
    }, "")
    return names

    // No Author
  } else {
    return "Artsy Editors"
  }
}

export const getDate = (date, layout = '') => {
  switch (layout) {
    case "monthYear": return moment(date).tz("America/New_York").format("MMMM YYYY")
    case "condensed": return moment(date).tz("America/New_York").format("MMM D, YYYY")
    default: return moment(date).tz("America/New_York").format("MMM D, YYYY h:mm a")
  }
}

export const getMediaDate = (article) => {
  const { published_at, scheduled_publish_at, media } = article
  const { release_date } = media

  if (release_date) {
    return release_date
  } else {
    return published_at || scheduled_publish_at
  }
}

export const formatTime = (time) => {
  let minutes = Math.floor(time / 60) % 60
  let seconds = Math.floor(time % 60)
  minutes = minutes <= 0 ? 0 : minutes
  seconds = seconds <= 0 ? 0 : seconds

  const minutesStr = minutes < 10 ? "0" + minutes : minutes
  const secondsStr = seconds < 10 ? "0" + seconds : seconds
  return minutesStr + ":" + secondsStr
}
