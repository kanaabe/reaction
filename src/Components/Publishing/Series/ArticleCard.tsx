import React, { Component } from "react"
import styled, { StyledFunction } from "styled-components"
import { crop } from "../../../Utils/resizer"
import { track } from "../../../Utils/track"
import { pMedia } from "../../Helpers"
import { Date } from '../Byline/AuthorDate'
import { Byline } from '../Byline/Byline'
import {
  formatTime,
  getMediaDate
} from "../Constants"
import { Fonts } from "../Fonts"
import { IconVideoPlay } from "../Icon/IconVideoPlay"

interface Props {
  article?: any
  color?: string
  editing?: boolean
  editDate?: any
  editDescription?: any
  editTitle?: any
  editImage?: any
  series?: any
  tracking?: any
}

@track()
export class ArticleCard extends Component<Props, null> {
  public static defaultProps: Partial<Props>

  isUnpublishedMedia() {
    const { media } = this.props.article
    return media && !media.published
  }

  isEditing = () => {
    const {
      editDate,
      editDescription,
      editImage,
      editTitle,
      editing
    } = this.props

    return editing || editDate || editDescription || editImage || editTitle
  }

  renderDate = () => {
    const { article, color, editDate } = this.props
    const { media } = article

    if (editDate) {
      return (
        <MediaDate>{editDate}</MediaDate>
      )
    } else if (media) {
      return this.renderMediaDate()
    } else {
      return (
        <Byline
          article={article}
          color={color}
          layout='condensed'
        />
      )
    }
  }

  renderMediaDate = () => {
    const { article } = this.props
    const mediaDate = getMediaDate(article)

    if (this.isUnpublishedMedia()) {
      return (
        <MediaDate>
          <span>Available </span>
          <Date layout="monthYear" date={mediaDate} />
        </MediaDate>
      )
    } else {
      return <Date layout='condensed' date={mediaDate} />
    }
  }

  renderMediaCoverInfo = () => {
    const { article, color } = this.props

    if (this.isUnpublishedMedia()) {
      return <MediaComingSoon>Coming Soon</MediaComingSoon>
    } else {
      return (
        <MediaPlay>
          <IconVideoPlay color={color} />
          {formatTime(article.media.duration)}
        </MediaPlay>
      )
    }
  }

  openLink = (e) => {
    e.preventDefault()

    if (!this.isUnpublishedMedia() && !this.isEditing()) {
      window.open(e.currentTarget.attributes.href.value)
    }

    this.props.tracking.trackEvent({
      action: "Click",
      label: "Related article card"
    })
  }

  render() {
    const {
      article,
      color,
      editDescription,
      editImage,
      editTitle,
      series
    } = this.props
    const { media } = article
    const isUnpublishedMedia = this.isUnpublishedMedia()

    return (
      <ArticleCardContainer
        href={isUnpublishedMedia ? '' : article.slug}
        color={color}
        className="ArticleCard"
        published={!isUnpublishedMedia}
        onClick={this.openLink}
      >

        <TextContainer>
          <div>
            <Header>
              <div>{series && series.title}</div>
            </Header>
            <Title>
              {editTitle
                ? editTitle
                : article.title
              }
            </Title>
            <Description>
              {editDescription
                ? editDescription
                : article.description
              }
            </Description>
          </div>
          {this.renderDate()}
        </TextContainer>

        <ImageContainer>
          {editImage
            ? editImage
            : <Image src={crop(article.thumbnail_image, { width: 680, height: 450 })} />
          }
          {media && this.renderMediaCoverInfo()}
        </ImageContainer>

      </ArticleCardContainer>
    )
  }
}

ArticleCard.defaultProps = {
  color: 'black'
}

interface LinkProps {
  published: boolean
}

const A: StyledFunction<Props & LinkProps & React.HTMLProps<HTMLLinkElement>> = styled.a


const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  transition: opacity 0.15s;
  display: block;
`

const ImageContainer = styled.div`
  position: relative;
  width: 50%;
  height: 100%;
  margin-left: 30px;
  ${props => pMedia.md`
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  `}
`

export const ArticleCardContainer = A`
  display: block;
  border: 1px solid;
  border-radius: 2px;
  color: ${props => props.color};
  cursor: ${props => props.published ? "pointer" : "default"};
  text-decoration: none;
  padding: 30px;
  display: flex;
  ${Image} {
    opacity: ${props => props.published ? "1" : "0.7"};
  }
  &:hover {
    ${Image} {
      opacity: .7;
    }
  }
  ${ImageContainer} {
    background: ${props => props.color === "white" ? "black" : "white"}
  }
  ${props => pMedia.md`
    flex-direction: column-reverse;
    padding: 20px;
  `}
`

const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
  .author, .date {
    ${Fonts.unica("s16", "medium")}
  }
  ${props => pMedia.md`
    width: 100%;
    margin-bottom: 0;
  `}
`

const Title = styled.div`
  ${Fonts.unica("s45")}
  margin-bottom: 30px;
  ${props => pMedia.md`
    ${Fonts.unica("s32")}
  `}
`

const Header = styled.div`
  ${Fonts.unica("s16", "medium")}
  margin-bottom: 10px;
`

const Description = styled.div`
  ${Fonts.garamond("s23")}
  ${props => pMedia.md`
    ${Fonts.garamond("s19")}
    margin-bottom: 20px;
  `}
`

const MediaDate = styled.div`
  ${Fonts.unica("s16", "medium")}
  display: flex;
  align-items: flex-end;
  span {
    margin-right: 5px;
  }
`

const Media = styled.div`
  position: absolute;
  left: 20px;
  bottom: 20px;
  right: 20px;
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
`

const MediaPlay = Media.extend`
  ${Fonts.unica("s16", "medium")}
  svg {
    width: 40px;
  }
  ${props => pMedia.md`
    svg {
      width: 30px;
    }
  `}
`

const MediaComingSoon = Media.extend`
  ${Fonts.unica("s45")}
  ${props => pMedia.md`
    ${Fonts.unica("s32")}
  `}
`
