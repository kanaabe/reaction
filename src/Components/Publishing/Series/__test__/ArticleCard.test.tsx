import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { getDate } from '../../Constants'
import { SeriesArticle, StandardArticle, VideoArticle } from "../../Fixtures/Articles"
import { EditableChild } from "../../Fixtures/Helpers"
import { IconVideoPlay } from "../../Icon/IconVideoPlay"
import { ArticleCard, ArticleCardContainer } from "../ArticleCard"

jest.mock('../../../../Utils/track.ts', () => ({
  track: jest.fn()
}))

describe("ArticleCard", () => {
  let standardArticle
  let videoArticle

  beforeEach(() => {
    videoArticle = VideoArticle
    standardArticle = StandardArticle
  })

  it("renders an article properly", () => {
    const component = renderer.create(
      <ArticleCard
        article={standardArticle}
        series={SeriesArticle}
      />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders an article with unpublished media properly", () => {
    const component = renderer.create(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
      />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders an article with children properly", () => {
    const component = renderer.create(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
        editDate={EditableChild('date')}
        editDescription={EditableChild('description')}
        editImage={EditableChild('image')}
        editTitle={EditableChild('title')}
      />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it("Renders media duration and play icon if article has media and is published", () => {
    videoArticle.media.published = true
    const component = mount(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
      />
    )

    expect(component.find(IconVideoPlay).length).toBe(1)
    expect(component.text()).toMatch('02:28')
  })

  it("Renders coming soon and available date if article has media and is unpublished", () => {
    videoArticle.media.published = false
    const component = mount(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
      />
    )

    expect(component.find(IconVideoPlay).length).toBe(0)
    expect(component.text()).not.toMatch('03:12')
    expect(component.text()).toMatch('Coming Soon')
    expect(component.text()).toMatch('Available ')
  })

  it("Does not render byline if article has media", () => {
    const component = mount(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
      />
    )

    expect(component.find('.author').length).toBe(0)
  })

  it("Renders the media release_date if present", () => {
    const component = mount(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
      />)
    const renderedDate = component.find('.date').first().text()
    const formattedDate = getDate(videoArticle.media.release_date, 'monthYear')

    expect(renderedDate).toBe(formattedDate)
  })

  it("Renders the published_at date if no release_date and media is present", () => {
    delete videoArticle.media.release_date
    const component = mount(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
      />
    )
    const renderedDate = component.find('.date').first().text()
    const formattedDate = getDate(videoArticle.published_at, 'monthYear')

    expect(renderedDate).toBe(formattedDate)
  })

  it("Renders editable fields if present", () => {
    const component = mount(
      <ArticleCard
        article={videoArticle}
        series={SeriesArticle}
        editDate={EditableChild('date')}
        editDescription={EditableChild('description')}
        editImage={EditableChild('image')}
        editTitle={EditableChild('title')}
      />
    )
    expect(component.text()).toMatch("Child date")
    expect(component.text()).toMatch("Child description")
    expect(component.text()).toMatch("Child image")
    expect(component.text()).toMatch("Child title")
  })

  describe("Analytics", () => {
    it("tracks article card click", () => {
      const trackEvent = jest.fn()
      const component = mount(
        <ArticleCard
          article={videoArticle}
          series={SeriesArticle}
          tracking={{
            trackEvent
          }}
        />
      )
      component.find(ArticleCardContainer).first().simulate("click")
      expect(trackEvent).toBeCalledWith({
        action: "Click",
        label: "Related article card"
      })
    })
  })
})
