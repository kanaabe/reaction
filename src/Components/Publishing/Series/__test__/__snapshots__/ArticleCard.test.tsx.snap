// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders an article properly 1`] = `
.c7 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin: 10px 30px 0 0;
}

.c7.date {
  white-space: nowrap;
}

.c6 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin: 10px 30px 0 0;
}

.c6.date {
  white-space: nowrap;
}

.c6:before {
  content: "";
  display: inline-block;
  min-width: 10px;
  min-height: 10px;
  border-radius: 50%;
  margin: 6px 10px 1px 0;
  background-color: black;
}

.c6:before {
  min-width: 8px;
  min-height: 8px;
  margin: 0 5px 1px 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: black;
}

.c11 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c9 {
  position: relative;
  width: 50%;
  height: 100%;
  margin-left: 30px;
}

.c0 {
  display: block;
  border: 1px solid;
  border-radius: 2px;
  color: black;
  cursor: pointer;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c10 {
  opacity: 1;
}

.c0:hover .c10 {
  opacity: .7;
}

.c0 .c8 {
  background: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c1 .author,
.c1 .date {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  margin-bottom: 10px;
}

.c4 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
}

@media (max-width:720px) {
  .c7 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 10px 20px 0 0;
  }
}

@media (max-width:720px) {
  .c6 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 10px 20px 0 0;
  }
}

@media (max-width:720px) {
  .c6:before {
    min-width: 8px;
    min-height: 8px;
  }
}

@media (max-width:900px) {
  .c9 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
    margin-bottom: 20px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="new-yorks-next-art-district"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2"
      >
        <div>
          The Future of Art
        </div>
      </div>
      <div
        className="c3"
      >
        New York's Next Art District
      </div>
      <div
        className="c4"
      >
        Land exhibitions, make influential contacts, and gain valuable feedback about your work.
      </div>
    </div>
    <div
      className="Byline c5"
      color="black"
    >
      <div
        className="author c6"
        color="black"
      >
        By 
        Casey Lesser
      </div>
      <div
        className="date c7"
      >
        May 19, 2017
      </div>
    </div>
  </div>
  <div
    className="c8 c9"
  >
    <img
      className="c10 c11"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F7lsxxsw0qPAuKl37jEYitw%252Farticle%2Basset%2B1-hig%2Bres%2Bcopy.jpg&width=680&height=450&quality=95"
    />
  </div>
</a>
`;

exports[`ArticleCard renders an article with children properly 1`] = `
.c9 {
  width: 32px;
  height: 32px;
}

.c7 {
  position: relative;
  width: 50%;
  height: 100%;
  margin-left: 30px;
}

.c0 {
  display: block;
  border: 1px solid;
  border-radius: 2px;
  color: black;
  cursor: pointer;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .file__Image-imvfnf-0 {
  opacity: 1;
}

.c0:hover .file__Image-imvfnf-0 {
  opacity: .7;
}

.c0 .c6 {
  background: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c1 .author,
.c1 .date {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  margin-bottom: 10px;
}

.c4 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
}

.c5 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 span {
  margin-right: 5px;
}

.c8 {
  position: absolute;
  left: 20px;
  bottom: 20px;
  right: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c8 svg {
  width: 40px;
}

@media (max-width:900px) {
  .c7 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
    margin-bottom: 20px;
  }
}

@media (max-width:900px) {
  .c8 svg {
    width: 30px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="joanne-artman-gallery-poetry-naturerefinement-form"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2"
      >
        <div>
          The Future of Art
        </div>
      </div>
      <div
        className="c3"
      >
        <div>
          Child 
          title
        </div>
      </div>
      <div
        className="c4"
      >
        <div>
          Child 
          description
        </div>
      </div>
    </div>
    <div
      className="c5"
    >
      <div>
        Child 
        date
      </div>
    </div>
  </div>
  <div
    className="c6 c7"
  >
    <div>
      Child 
      image
    </div>
    <div
      className="c8"
    >
      <svg
        className="c9"
        version="1.1"
        viewBox="0 0 40 56"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="black"
            points="0 0 0 56 39.3600006 28"
          />
        </g>
      </svg>
      02:28
    </div>
  </div>
</a>
`;

exports[`ArticleCard renders an article with unpublished media properly 1`] = `
.c11 {
  width: 32px;
  height: 32px;
}

.c5 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin: 10px 30px 0 0;
}

.c5.date {
  white-space: nowrap;
}

.c9 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c7 {
  position: relative;
  width: 50%;
  height: 100%;
  margin-left: 30px;
}

.c0 {
  display: block;
  border: 1px solid;
  border-radius: 2px;
  color: black;
  cursor: pointer;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .c8 {
  opacity: 1;
}

.c0:hover .c8 {
  opacity: .7;
}

.c0 .c6 {
  background: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c1 .author,
.c1 .date {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c3 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
  margin-bottom: 10px;
}

.c4 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 23px;
  line-height: 1.5em;
}

.c10 {
  position: absolute;
  left: 20px;
  bottom: 20px;
  right: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c10 svg {
  width: 40px;
}

@media (max-width:720px) {
  .c5 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 10px 20px 0 0;
  }
}

@media (max-width:900px) {
  .c7 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c1 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c3 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c4 {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 19px;
    line-height: 1.5em;
    margin-bottom: 20px;
  }
}

@media (max-width:900px) {
  .c10 svg {
    width: 30px;
  }
}

<a
  className="ArticleCard c0"
  color="black"
  href="joanne-artman-gallery-poetry-naturerefinement-form"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2"
      >
        <div>
          The Future of Art
        </div>
      </div>
      <div
        className="c3"
      >
        Trevor Paglan
      </div>
      <div
        className="c4"
      >
        The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
      </div>
    </div>
    <div
      className="date c5"
    >
      Aug 28, 2017
    </div>
  </div>
  <div
    className="c6 c7"
  >
    <img
      className="c8 c9"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FwHFgQlrTrHav5O6bQRJ0dg%252FUntitled%2BSuspended_30x67x33%2B%25282%2529_sm%2Bcropped.jpg&width=680&height=450&quality=95"
    />
    <div
      className="c10"
    >
      <svg
        className="c11"
        version="1.1"
        viewBox="0 0 40 56"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="black"
            points="0 0 0 56 39.3600006 28"
          />
        </g>
      </svg>
      02:28
    </div>
  </div>
</a>
`;
